include/opentracing/dynamic_load.h

```diff
@@ -35,14 +36,45 @@
 //      assert(error_category != nullptr);
 //      std::error_code error{rcode, *error_category};
 //   }
-extern "C" {
-#ifdef OPENTRACING_BUILD_DYNAMIC_LOADING
-int __attribute((weak))
-OpenTracingMakeTracerFactory(const char* opentracing_version,
-                             const void** error_category,
-                             void** tracer_factory);
+
+using OpenTracingMakeTracerFactoryType = int(const char* opentracing_version,
+                                        const void** error_category,
+                                        void** tracer_factory);
+
+#ifdef WIN32
+
+#define OPENTRACING_DECLARE_IMPL_FACTORY(X)                                 \
+  extern "C" {                                                              \
+  \
+__declspec(dllexport) int X(const char* opentracing_version,                \
+                            const void** error_category,                    \
+                            void** tracer_factory);                         \
+  \
+extern __declspec(dllexport)                                                \
+      OpenTracingMakeTracerFactoryType* const OpenTracingMakeTracerFactory; \
+  \
+__declspec(selectany) OpenTracingMakeTracerFactoryType* const               \
+      OpenTracingMakeTracerFactory = X;                                     \
+  }  // extern "C"
+
+#else
+
+#define OPENTRACING_DECLARE_IMPL_FACTORY(X)                                 \
+  extern "C" {                                                              \
+                                                                            \
+  int __attribute((weak))                                                   \
+      X(const char* opentracing_version, const void** error_category,       \
+        void** tracer_factory);                                             \
```

Comment:
I don't think we need these declarations:

```cpp
+__declspec(dllexport) int X(const char* opentracing_version,                \
+                            const void** error_category,                    \
+                            void** tracer_factory);
```
```cpp
+  int __attribute((weak))                                                   \
+      X(const char* opentracing_version, const void** error_category,       \
+        void** tracer_factory); 
```
All that needs to be exposed is the variable `OpenTracingMakeTracerFactory`. `X` can be a static function in the translation unit.

Reply:
This declaration is useful so that 
```
OPENTRACING_DECLARE_IMPL_FACTORY(OpenTracingMakeTracerFactoryFct);
```
can be declared anywhere in the translation unit.

I'll do the necessary changes.